### ISSUE TITLE
correctness: a constraint on dm.sum(X, axis=k) is certified optimal at the axis-collapsed answer (false bound)

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/discopt#1160]
## Summary

A constraint whose body is an **axis-reduced** sum, `dm.sum(X, axis=k)`, is solved as if the reduction were over *every* axis. The solver returns `status="optimal"` with a dual bound that **excludes the true optimum**, on a model whose own NLP evaluator handles the rows correctly.

Found while building a Python-side differential panel for jkitchin/discopt#1158; it is **pre-existing and unrelated to that PR** — measured identically on `1e75919` (`main`) and on the PR head, so nothing in #1158 causes or fixes it.

## Reproducer

```python
import numpy as np
import discopt.modeling.core as dm
from discopt.validation.feasibility import verify_point

m = dm.Model("sum_axis")
A = m.continuous("A", shape=(2, 3), lb=0, ub=1)
m.subject_to(dm.sum(A, axis=1) <= 2)     # per-ROW cap
m.minimize(-dm.sum(A))                   # maximize the total

r = m.solve(time_limit=60, gap_tolerance=1e-6)
print(r.status, r.objective, r.bound)
# -> optimal -2.0000000 -2.0000000

# ... but this point is feasible and twice as good:
better = np.array([1., 1., 0., 1., 1., 0.])   # row sums 2 and 2
print(verify_point(m, better, with_objective=True))
# -> VerifyResult(ok=True, objective=-4.0, reason='')
```

Each row sums to at most 2, so the total is at most 4 and the true optimum is **-4**. The solver certifies **-2**, and reports it as a *dual bound* — for a minimize, a valid lower bound must be `<= -4`, so `-2` is above the optimum and fathoms it away. `incorrect_count` would count this row.

## What is and is not at fault

The NLP evaluator is **correct** — it is not a row-mapping problem:

```
evaluator n_constraints: 2            (expected 2 rows: one per row of A)
row map:                [(0, 2, 'rowcap')]
g(x) at the true optimum [1,1,0,1,1,0]: [0. 0.]     (both rows satisfied)
```

The failure is downstream, in the relaxation/branch-and-bound path. The returned value is exactly the answer of the **axis-collapsed** model:

```python
m2 = dm.Model("total")
B = m2.continuous("B", shape=(2, 3), lb=0, ub=1)
m2.subject_to(dm.sum(B) <= 2)          # TOTAL cap, not per-row
m2.minimize(-dm.sum(B))
m2.solve(...).objective                # -> -2.000000, the same wrong answer
```

So something in the relaxation build treats `sum(X, axis=k)` as a full reduction. `_relax/dag_compiler.py:340` lowers the node correctly (`jnp.sum(a[0], axis=expr.axis)`), and `_relax/convexity/interval_eval.py` now reduces along `expr.axis` too (fixed in #1158) — neither is the remaining culprit. The likely places to look are the other `SumExpression` handlers that descend into `.operand` without consulting `.axis`, which is sound for a full reduction and not for an axis one:

- `_relax/milp_relaxation.py:1389` and `:1478`
- `_relax/least_squares.py:226`
- `_relax/obbt.py:501`

(Those four were reviewed in #1158 and cleared *as structural pattern-matchers* — term collectors and coefficient extractors, where descending is legitimate because sum is linear. That reasoning holds for `axis=None`; whether it still holds when the node reduces only one axis of a multi-dimensional operand is exactly the question to answer here.)

## Scope

Axis-reduced sums only. Both of these are correct today:

```
flat 1-D, scalar row      min -sum(B) s.t. sum(B) <= 4, B in [0,1]^6      -> -4  correct
explicit per-row terms    min -sum(B) s.t. B[0,0]+B[0,1]+B[0,2] <= 2      -> -5  correct
axis-reduced row          min -sum(A) s.t. sum(A, axis=1) <= 2            -> -2  WRONG (true -4)
```

It reaches models built through the Python modeling API; the `.nl` reader does not emit `SumExpression` at all (measured: 0 of the 66 `.nl` files in `python/tests/data/minlplib_nl/`), so no `.nl` panel or gate covers this class — which is presumably why it has gone unnoticed.

## Done looks like

1. Identify which relaxation-side handler ignores `SumExpression.axis` (the four call sites above are the candidates), and fix it there rather than at a call site.
2. A regression test that fails before and passes after, asserting **both** that the objective is `-4` and that `bound <= -4 + tol`. Assert the bound, not just the objective: the objective alone would pass on a build that merely got lucky on the primal.
3. Extend it to at least one nonlinear axis-reduced case (e.g. `sum(A * A, axis=1) <= c`), since the linear one may be fixed by a term collector that the nonlinear path does not share.
4. Add a `dm.sum(..., axis=k)` model to whichever Python-API correctness panel exists, since the `.nl` corpus structurally cannot cover it.
